### ISSUE TITLE
Inherit Swing Terminal for child process

### ISSUE THREAD

**Issue by rvansa** [mabe02/lanterna#449]
Hi, is it possible to make the Swing terminal display output from a child process? E.g. I want to start `vi` or `less` using `new ProcessBuilder(...).inheritIO()` while still developing using the Swing window. It works fine if I force Lanterna to use the regular terminal, but for development it would be comfier to stick with Swing.

Thanks

**Comment by avl42** [comment]
First of all, the SwingTerminal  is not a "real" terminal.   It is a Java
class that implements the API that Lanterna's TerminalScreen class uses. If
you use a Lanterna-app in e.g. gnome-terminal or putty, then Lanterna uses
a different "Terminal" implementation, that will communicate with the real
terminal, and if you start a child-process, then the child-process will
talk directly to that real terminal, not to Lanterna's
"real-Terminal"-driver.

If you absolutely *need* the child-process to output to the Swing-terminal,
then you'll need to capture the process' I/O and redirect them to methods
of SwingTerminal.  This is still relatively easy (though not trivial) for
programs merely writing progress/log messages to stdout, and it is
highly-complicated and platform-specific if you then also want to interpret
escape-sequences, and providing a pseudo-tty to the child-process (as you
would need for "vi").

in a nutshell: nothing is impossible, if you're willing to write *lots* of
extra code for it.

**Comment by rvansa** [comment]
@avl42 Thanks for the elaborate response - I wasn't sure about the complexity of effectively doing the terminal in Swing, I just knew that I am not going to do that myself :) Given that you've confirmed that it is 'highly-complicated', I'll consider this request for enhancement a 'Won't do'.
Cheers and keep up the good work.
